plugins/repository-hdfs/hadoop-client-api/build.gradle

```diff
@@ -27,16 +29,23 @@ def patchTask = tasks.register("patchClasses", JavaExec) {
   outputs.dir(outputDir)
   classpath = sourceSets.patcher.runtimeClasspath
   mainClass = 'org.elasticsearch.hdfs.patch.HdfsClassPatcher'
+  def thejar = configurations.thejar
   doFirst {
-    args(configurations.thejar.singleFile, outputDir.get().asFile)
```

Comment:
we cannot reference `configurations` in the execution phase for cc compatibility